Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/DimensionSpacePointsRepository.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentGraph.DoctrineDbalAdapter package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository;
+
+use Doctrine\DBAL\Connection;
+use Neos\ContentRepository\Core\DimensionSpace\AbstractDimensionSpacePoint;
+use Neos\ContentRepository\Core\DimensionSpace\OriginDimensionSpacePoint;
+
+/**
+ * A place to cache dimensionspacepoint data, it's deterministic so there should never be an issue with having this in memory.
+ *
+ * @internal you should never need this in userland code
+ */
+final class DimensionSpacePointsRepository
+{
+    /**
+     * @var array<string, string>
+     */
+    private array $dimensionspacePointsRuntimeCache = [];
+
+    public function __construct(
+        private readonly Connection $databaseConnection,
+        private readonly string $tableNamePrefix
+    ) {
+    }
+
+    public function insertDimensionSpacePoint(AbstractDimensionSpacePoint $dimensionSpacePoint): void
+    {
+        if ($this->getCoordinatesByHashFromRuntimeCache($dimensionSpacePoint->hash) !== null) {
+            return;
+        }
+
+        $this->dimensionspacePointsRuntimeCache[$dimensionSpacePoint->hash] = $dimensionSpacePoint->toJson();
+        $this->writeDimensionSpacePoint($dimensionSpacePoint->hash, $dimensionSpacePoint->toJson());
+    }
+
+    /**
+     * @param string $hash
+     * @param array<string,string> $dimensionSpacePointCoordinates
+     * @return void
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function insertDimensionSpacePointByHashAndCoordinates(string $hash, array $dimensionSpacePointCoordinates): void
+    {
+        if ($this->getCoordinatesByHashFromRuntimeCache($hash) !== null) {
+            return;
+        }
+
+        $dimensionSpacePointCoordinatesJson = json_encode($dimensionSpacePointCoordinates, JSON_THROW_ON_ERROR);
+        $this->dimensionspacePointsRuntimeCache[$hash] = $dimensionSpacePointCoordinatesJson;
+        $this->writeDimensionSpacePoint($hash, $dimensionSpacePointCoordinatesJson);
+    }
+
+    public function getOriginDimensionSpacePointByHash(string $hash): OriginDimensionSpacePoint
+    {
+        $coordinates = $this->getCoordinatesByHashFromRuntimeCache($hash);
+        if ($coordinates === null) {
+            $this->fillRuntimeCacheFromDatabase();
+            $coordinates = $this->getCoordinatesByHashFromRuntimeCache($hash);
+        }
```

Comment:
ha this is a bit hacky as we dont pass this repository as singleton that means it cannot share its runtime cache and would otherwise not pick up on new OriginDimensionSpacePoints, but i consider this a very safe runtime cache as nothing can go wrong (despite it being outdated, in which case we refetch)?

Reply:
true, we _could_ even make this static if we wanted LOL

Reply:
nah its fine thx, i guess asking the db like this is good enough and it will also work for async / parallel